eth/peer.go

```diff
@@ -230,10 +231,12 @@ func (p *peer) RequestReceipts(hashes []common.Hash) error {
 
 // Handshake executes the eth protocol handshake, negotiating version number,
 // network IDs, difficulties, head and genesis blocks.
-func (p *peer) Handshake(network uint64, td *big.Int, head common.Hash, genesis common.Hash) error {
+func (p *peer) Handshake(network uint64, chainconfig *params.ChainConfig, status *statusData) error {
```

Comment:
Why did this change? And if this changed, are there no usages of `Handshake` that also need to be changed?

Reply:
That change was to send a fake status message that says our node is synced up to the DAO fork block. As you said in the comment above, what we advertise in our Status message shouldn't affect peer connections, so I might revert this change. One benefit we get from hardcoding Status info, however, is that the client no longer needs to check its local blockchain state every time it sends the message.